http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/response/CrtResponseAdapter.java

```diff
@@ -126,7 +126,7 @@ private void onSuccessfulResponseComplete(HttpStream stream) {
 
     private void handlePublisherError(HttpStream stream, Throwable failure) {
         failResponseHandlerAndFuture(stream, failure);
-        responseHandlerHelper.releaseConnection(stream);
+        responseHandlerHelper.closeConnection(stream);
```

Comment:
Changed this from release to close because @TingDaoK pointed to me that the connection should be shut down instead of releasing back to the pool in this case.